File: abstractions/typescript/src/store/inMemoryBackingStoreFactory.ts
import { BackingStore } from "./backingStore";
import { BackingStoreFactory } from "./backingStoreFactory";
import { InMemoryBackingStore } from "./inMemoryBackingStore";

/** This class is used to create instances of InMemoryBackingStore */
export class InMemoryBackingStoreFactory implements BackingStoreFactory {
    public createBackingStore(): BackingStore {
        return new InMemoryBackingStore();
    }
}

File: abstractions/python/kiota/abstractions/store/in_memory_backing_store_factory.py
from .backing_store import BackingStore
from .backing_store_factory import BackingStoreFactory
from .in_memory_backing_store import InMemoryBackingStore


class InMemoryBackingStoreFactory(BackingStoreFactory):
    """This class is used to create instances of InMemoryBackingStore
    """

    def create_backing_store(self) -> BackingStore:
        return InMemoryBackingStore()
